homeassistant/components/switch/amcrest.py

```diff
@@ -38,11 +38,11 @@ def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
 class AmcrestSwitch(ToggleEntity):
     """Representation of an Amcrest IP camera switch."""
 
-    def __init__(self, setting, camera):
```

Comment:
This can't work, since you don't pass the name variable when calling init.

Reply:
Odd I thought my submission changed this line to pass the name